vmware-base/src/com/cloud/hypervisor/vmware/mo/VirtualMachineMO.java

```diff
@@ -2118,8 +2118,12 @@ public int getScsiDiskControllerKey(String diskController) throws Exception {
         List<VirtualDevice> devices = (List<VirtualDevice>)_context.getVimClient().
                 getDynamicProperty(_mor, "config.hardware.device");
 
+        String deviceList = "";
```

Comment:
Why not represent as an actual ``List`` and use ``Joiner`` to render it to a ``String``?